src/client/features/interactions/interactions-edge-tooltip.js

```diff
@@ -10,8 +10,17 @@ class InteractionsEdgeTooltip extends React.Component {
     super(props);
 
     this.state = {
+      datasources: [],
       publications: []
     };
+
+    // Update datasources content to get homepage?
+    ServerAPI.datasources()
```

Comment:
Why can't you send this request once at the time the network is generated and compute the datasources for all elements there?